x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -297,11 +297,11 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
     [form]
   );
 
-  const [aggFields, setAggregatableFields] = useState<DataViewFieldBase[]>([]);
+  const [aggFields, setAggregatableFields] = useState<BrowserField[]>([]);
 
   useEffect(() => {
     const { fields } = indexPattern;
-    setAggregatableFields(fields);
+    setAggregatableFields(aggregatableFields(fields as BrowserField[]));
```

Comment:
`DataViewBase`, the type of `fields`, doesn't contain the `aggregatable` property. Is that type incorrect? Otherwise, it wouldn't work with `aggregatableFields`.

Reply:
Yeah the types are incorrect unfortunately.

Reply:
Yeah, I checked the types, and it looks like at least `indexPattern` and `aggFields` are incorrectly typed. So the data we receive and work with on the rule creation page doesn't match the declared interfaces. @dhurley14 Do you know where the wrong types come from? 

It's a bit concerning that the rule creation page is poorly typed. It would be good to leave a comment warning that types are wrong, or even better, find the root cause and fix it. Do we have a ticket for that?

cc @banderror

Reply:
I can update the code and leave a comment about the types. I'm not exactly sure as to where the types are coming from unfortunately.